crates/milli/src/search/new/tests/geo_sort.rs

```diff
@@ -95,6 +95,47 @@ fn test_geo_sort() {
     insta::assert_snapshot!(format!("{scores:#?}"));
 }
 
+#[test]
+fn test_geo_sort_with_following_ranking_rules() {
+    let index = create_index();
+
+    index
+        .add_documents(documents!([
+            { "id": 1 }, { "id": 4 }, { "id": 3 }, { "id": 2 }, { "id": 5 },
+            { "id": 6, RESERVED_GEO_FIELD_NAME: { "lat": 2, "lng": 2 }, "score": 10 },
+            { "id": 7, RESERVED_GEO_FIELD_NAME: { "lat": 2, "lng": 2 }, "score": 9 },
+            { "id": 8, RESERVED_GEO_FIELD_NAME: { "lat": 2, "lng": 2 }, "score": 8 },
+            { "id": 9, RESERVED_GEO_FIELD_NAME: { "lat": 2, "lng": 2 }, "score": 7 },
+            { "id": 10, RESERVED_GEO_FIELD_NAME: { "lat": 2, "lng": 2 }, "score":6 },
+            { "id": 11, RESERVED_GEO_FIELD_NAME: { "lat": 2, "lng": 2 }, "score": 5 },
+            { "id": 12, RESERVED_GEO_FIELD_NAME: { "lat": 5, "lng": 5 }, "score": 10 },
+            { "id": 13, RESERVED_GEO_FIELD_NAME: { "lat": 5, "lng": 5 }, "score": 9 },
+            { "id": 14, RESERVED_GEO_FIELD_NAME: { "lat": 5, "lng": 5 }, "score": 8 },
+            { "id": 15, RESERVED_GEO_FIELD_NAME: { "lat": 5, "lng": 5 }, "score": 7 },
+        ]))
+        .unwrap();
+
+    let rtxn = index.read_txn().unwrap();
+
+    let mut s = Search::new(&rtxn, &index);
+    s.scoring_strategy(crate::score_details::ScoringStrategy::Detailed);
+    s.sort_criteria(vec![
+        AscDesc::Asc(Member::Geo([0., 0.])),
+        AscDesc::Desc(Member::Field("score".to_string())),
+    ]);
+    let (ids, scores) = execute_iterative_and_rtree_returns_the_same(&rtxn, &index, &mut s);
+    insta::assert_snapshot!(format!("{ids:?}"), @"[6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 1, 4, 3, 2, 5]");
+    insta::assert_snapshot!(format!("{scores:#?}"));
+
+    s.sort_criteria(vec![
+        AscDesc::Desc(Member::Geo([0., 0.])),
+        AscDesc::Desc(Member::Field("score".to_string())),
+    ]);
+    let (ids, scores) = execute_iterative_and_rtree_returns_the_same(&rtxn, &index, &mut s);
+    insta::assert_snapshot!(format!("{ids:?}"), @"[12, 13, 14, 15, 6, 7, 8, 9, 10, 11, 1, 4, 3, 2, 5]");
+    insta::assert_snapshot!(format!("{scores:#?}"));
+}
+
```

Comment:
These tests are awesome.
I was just expecting two more test with a very small bucket of like 2 elements.
Let me know if it makes sense and if you can add them?

Then I think we’ll be good to merge 👌

Reply:
In fact, execute_iterative_and_rtree_returns_the_same covers both the cases where the bucket size is greater than the cache size (need to refill the cache) and less than the cache size (no refill is required). I wonder which case a very small bucket is for?

Reply:
Huum from what I see we have a test where the cache is small and must be refilled before we reach our thresholds. That’s good.
But I don’t see any test doing the opposite, basically, we have a very low threshold and must stop early even though the bucket is not empty

Reply:
Hi, @irevoire I hope you had a great holiday.

You wanted to test the case where max_bucket_size is reached, right? I think I misunderstood you earlier, I will add the corresponding tests later.

Reply:
Yes exactly!
Thank you 🙏 
Since there is two caches at the same time it’s not easy to explain ahah 😅